server/game/cards/Time/FlockShepherd.js

```diff
@@ -0,0 +1,38 @@
+const { BattlefieldTypes } = require('../../../constants.js');
+const AbilityDsl = require('../../abilitydsl.js');
+const Card = require('../../Card.js');
+
+class FlockShepherd extends Card {
+    setupCardAbilities(ability) {
+        //protect
+        this.addEffectToFlock(ability.effects.cannotBeDicePowerTarget());
+        this.addEffectToFlock(ability.effects.cannotBeSpellTarget());
+        this.addEffectToFlock(ability.effects.cannotBeAbilityTarget());
+
+        //gather
+        this.forcedReaction({
+            when: {
+                onAttackersDeclared: (event, context) =>
+                    event.attackingPlayer === context.source.owner.opponent
+            },
+            gameAction: AbilityDsl.actions.cardLastingEffect((context) => ({
+                target: context.player.unitsInPlay.filter((c) => c.printedAttack === 0),
+                effect: AbilityDsl.effects.modifyAttack(2),
+                duration: 'untilEndOfTurn'
+            })),
+            effect: 'increase the attack attack value of units with a printed attack of 0 by 2'
+        });
+    }
+
+    addEffectToFlock(effect) {
```

Comment:
You can pass an array of effects to the effect: property in persistentEffect to make this simpler (e.g. iron rhino)

Reply:
ah cool ill do that everywhere

Reply:
fixed